src/ui/component/dateInput/dateInput.tsx

```diff
@@ -0,0 +1,55 @@
+import { useMemo, type FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Icon } from '@/ui/component/icon/icon'
+import './dateInput.scss'
+import type { PartialDateTimeFormatOptions } from '@/util/date/date'
+import { formatDateToPattern } from '@/util/date/date'
+
+type DateInputProps = {
+  date?: Date
+}
+
+export const DateInput: FC<DateInputProps> = ({ date }) => {
+  const { i18n } = useTranslation()
+  const locale = i18n.language
+
+  const pattern = useMemo(
+    () => ({
+      day: 'DD',
+      month: 'MM',
+      year: 'YY'
+    }),
+    []
+  )
+
+  const dateFormatOptions: PartialDateTimeFormatOptions = useMemo(
+    () => ({
+      day: '2-digit',
+      month: '2-digit',
+      year: 'numeric'
+    }),
+    []
+  )
```

Comment:
Please, move these constants away from the React function to prevent from unnecessary `memo` that have a cost in memory usage.

Reply:
Oops, thanks for the feedback!